GenumKit/Parsers/AssetsCatalogParser.swift

```diff
@@ -8,24 +8,42 @@ import Foundation
 import PathKit
 
 public final class AssetsCatalogParser {
-  var imageNames = [String]()
+  var entries = [Entry]()
 
   public init() {}
 
   @discardableResult
   public func addImage(named name: String) -> Bool {
-    if imageNames.contains(name) {
+    if (entries.contains { $0.name == name }) {
       return false
     } else {
-      imageNames.append(name)
+      entries.append(Entry(name: name, value: name))
       return true
     }
   }
 
   public func parseCatalog(at path: String) {
     guard let items = loadAssetCatalog(at: path) else { return }
+
     // process recursively
-    processCatalog(items: items)
+    entries = process(items: items)
+  }
+
+  struct Entry {
+    var name: String
+    var value: String
+    var items: [Entry]?
```

Comment:
Haven't used enums with associated values yet (although I've read about them). I'll look into it.